src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              %   // Type backward reference
+    //              ?   // symbol backward reference
+    //              $   // identifier backward reference
```

Comment:
> $ is currently used where identifiers have a different meaning, i.e. LName in the grammar. Not sure it is easily replaceable

Not sure what you mean here.  The GNU as documentation states in a number of targets that you may not use `$` in symbol names.

Well, most should be OK - but you need a special case (i.e: `Target::dollars_in_identifiers`).

See here: https://sourceware.org/binutils/docs/as/Symbol-Intro.html#index-characters-used-in-symbols-130

> A symbol is one or more characters chosen from the set of all letters (both upper and lower case), digits and the three characters `_.$'.

So the only valid symbol characters are `[A-Za-z0-9_.$]` to start with.